src/jit/IntMath32Inl.h

```diff
@@ -611,14 +626,14 @@ static void emitBinary(sljit_compiler* compiler, Instruction* instr)
             opcode = SLJIT_XOR;
             break;
         case ByteCode::I32ShlOpcode:
-            opcode = SLJIT_SHL;
-            break;
+            emitShift32(compiler, SLJIT_SHL, args);
```

Comment:
There is a masked variant of these opcodes, e.g. SLJIT_MSHL:

https://github.com/zherczeg/sljit/blob/master/sljit_src/sljitLir.h#L1181